build.sh

```diff
@@ -111,45 +139,79 @@ inject_metadata () {
 # Main entry point.
 # ------------------------------------------------
 
-# Make sure we start from a clean slate.
-rm -rf built
-rm -rf downloads_cache
-rm -f master.tar.gz
-rm -f metadata.tar.gz
-
-# Run basic tests.
-echo "Running basic sanity tests on metadata."
-echo "If these fail, then fix whatever is causing them first."
+if [ -n "$ghprbActualCommit" ]
+then
+    echo "Commit hash: $ghprbActualCommit"
+    export COMMIT_CHANGES="`git diff --diff-filter=AM --name-only --stat origin/master`"
+else
+    echo "No commit provided, skipping further tests."
+    exit $EXIT_OK
+fi
 
-if ! prove
+# Make sure we start from a clean slate.
+if [ -d "built/" ]
 then
-    echo "Prove step failed."
-    exit 1
+    rm -rf built
 fi
 
-# Find the changes to test.
-echo "Finding changes to test..."
+if [ -d "downloads_cache/" ]
+then
+    rm -rf downloads_cache
+fi
 
-if [ -z $ghprbActualCommit ]
+if [ -e "master.tar.gz" ]
 then
-    echo "No commit hash, running all netkan files"
-    export COMMIT_CHANGES=NetKAN/*.netkan
-else
-    echo "Commit hash: $ghprbActualCommit"
-    export COMMIT_CHANGES="`git diff --diff-filter=AM --name-only --stat origin/master NetKAN`"
+    rm -f master.tar.gz
 fi
 
-if [ "$COMMIT_CHANGES" = "" ]
+if [ -e "metadata.tar.gz" ]
 then
-    echo "No .netkan changes, skipping further tests."
-    exit 0
+    rm -f metadata.tar.gz
 fi
 
 # Check JSON.
 echo "Running jsonlint on the changed files"
 echo "If you get an error below you should look for syntax errors in the metadata"
 
-jsonlint -s -v $COMMIT_CHANGES
+for f in $COMMIT_CHANGES
+do
+    if [ "$f" == "build.sh" ]
+    then
+        echo "Lets try not to validate our build script with jsonlint"
+        continue
+    fi
+
+    echo "Validating $f..."
+    jsonlint -s -v $f
+    
+    if [ $? -ne 0 ]
+    then
+        echo "Failed to validate $f"
+        exit $EXIT_FAILED_JSON_VALIDATION
+    fi
+done
+echo ""
+
```

Comment:
I moved the jsonlint above, it provides a more obvious message than we get from the metadata.t tests.